### PR TITLE
Whitesource remediate/gradle 8.x

### PR DESCRIPTION
**Pull request by reta**
### Description
Reopening https://github.com/opensearch-project/opensearch-sdk-java/pull/441 with JUnit Framework workaround

### Issues Resolved
See please https://github.com/opensearch-project/opensearch-sdk-java/pull/441

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dbwiddis I sadly cannot push changes to https://github.com/opensearch-project/opensearch-sdk-java/pull/441, had to reopen the pull request (with your changes included), thank you.

**Review by saratvemulapalli — APPROVED** [review]
Thank you @reta !
@dbwiddis would love your eyes as you looked at this earlier.

**Review by owaiskazi19 — APPROVED** [review]
Welcome to the SDK repo @reta!

**Review by dbwiddis — DISMISSED** [review]

**Comment by reta** [comment]
Build fails with 

```
ERROR StatusConsoleListener An exception occurred processing Appender console
 org.apache.logging.log4j.core.appender.AppenderLoggingException: java.lang.LinkageError: loader 'platform' attempted duplicate class definition for sun.text.resources.cldr.ext.FormatData_tzm. (sun.text.resources.cldr.ext.FormatData_tzm is in module jdk.localedata of loader 'platform')
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:165)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:134)
```

Looking into it ...

**Comment by dbwiddis** [comment]
Changes "look" good but not sure why the tests are failing so holding off on merging.

**Comment by dbwiddis** [comment]
For the record, 8.0.2 didn't fully fix the issue, so we are still waiting for 8.1 on this workaround.  @reta of note we use Junit 5.

**Comment by reta** [comment]
> For the record, 8.0.2 didn't fully fix the issue, so we are still waiting for 8.1 on this workaround. @reta of note we use Junit 5.

That's what I figured out, fixing that, thanks @dbwiddis :)

File: build.gradle [hunk 1]
```diff
@@ -84,12 +85,14 @@ dependencies {
...
     def junit4Version = "4.13.2"
     def junit5Version = "5.9.2"
+    def junitPlatform = "1.9.2"
     def jaxbVersion = "2.3.1"
     def requireJavadocVersion = "1.0.6"
 
     implementation("org.opensearch:opensearch:${opensearchVersion}")
     implementation("org.apache.logging.log4j:log4j-api:${log4jVersion}")
     implementation("org.apache.logging.log4j:log4j-core:${log4jVersion}")
+    implementation("org.apache.logging.log4j:log4j-jul:${log4jVersion}")
```
Comment (reta) [inline build.gradle:95]: This is unrelated change, but noticed  `log4j-jul` was 2.17.1

**Comment by reta** [comment]
@dbwiddis should be all set, thanks!

**Review by owaiskazi19 — APPROVED** [review]

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**